### DISCUSSION TITLE
VNCGUI 'enhancements'

### DISCUSSION THREAD

**Discussion by AstroKeith** [WimDeMeester/eFinder#57] [General]
I'm installing eFinder on a friends ServoCat driven 24" Dobsonian. He does astrophotography with the scope and so needs pin point accuracy to get the object on the sensor chip.

This made me look more carefully at the ServoCat interfacing. First of course we had to swap out the ArgoNavis for a Nexus DSC Pro.

Initially I left the ServoCat connection as recommended - a serial and a USB from the Nexus DSC to the ServoCat. Then connected the eFinder via wifi. I used a standalone wifi router in his observatory and set Nexus to infrastructure mode with a fixed IP.

That worked, but he really wanted remote control from his warm room. So ideally he would use SkySafari and the VNCGUI eFinder version. The Nexus DSC cant serve two devices over wifi. My first approach was to modify the VNCGUI so he could manually enter the RA & Dec of a target, upload it to the Nexus DSC and command a goto++. So SkySafari didnt need a wifi connection. An easy modification, useful, but not very efficient from a user POV!

I really wanted my Nexus DSC USB port back! (allowing SkySafari to have the wifi). The Nexus to ServoCat USB cable is only used to command GoTo's. So I've now removed it and reconnected the eFinder to the Nexus USB port. Now, a goto target can be initiated by either the Nexus DSC or the SkySafari, but it won't start. On the VNCGUI you hit the GoTo button, which reads and displays the set target on the GUI, and commands a GoTo++. SkySafari is happy and shows the expected live position of the scope.

This seems to be what the user needed. It's not been fully tested yet (only on test rig with a ServoCat emulator). But looks useful. I'll think about how to add it to the handbox version. There are a lot of eFinder users in the USA and they all have ServoCat!

While doing these modifications  I also changed....

Added the manual goto target input box. With an upload to Nexus button.
Set the GUI default text to bold. The Pi doesnt antialias properly and now the text is easier to read.
Moved the display of solve time into the box_write() area. 
Put readNexus() into a new thread with a 0.2 second repeat loop. Thus the GUI display is 'live'. Then removed all other calls to readNexus() as they arent necessary (I think!)
Inserted a try/except to catch to bug when Target is pressed before a solution is found.
Added a camera frame option to the eyepiece circle display options.

I can provide more details, or you may want me to finish field testing? Just let know.

**Comment by WimDeMeester** [comment]
Sorry for the very late reply, but I was enjoying holidays in Italy.  
I think I still miss something, because I don't understand the setup:
- The eFinder RaspBerry PI is connected using a USB cable to the Nexus DSC.  Is this correct?
- Is there any connection between the ServoCAT and the Nexus DSC at all?  
- The GoTo can only be done by pressing the 'GoTo' button in the GUI?  Does it also work using the handpad?

I'm in any case interested in the results of the field testing!

**Reply by AstroKeith** [reply]
Attached is a pdf showing the 'preferred' connections for this new method.
Normally the Nexus has a serial and a USB connection direct to the ServoCat. This would mean that eFinder can only access the Nexus via wifi. OK except then no other devices can connect to the Nexus ie SkySafari. This method however is being used by quite a few users in the USA.
The new user needed SkySafari. So for him a change was needed. The USB from Nexus to ServoCat carries only limited comms.

- Goto destination coordinates
- Goto stop command
- Move NSEW commands

So it was easy to add a new USB cable from the eFinder to the ServoCat to send these. I didnt need the Move, so have just done the goto's.
On the GUI variant, I can get the currently stored Nexus goto target (can be from its own catalogue or SKySafari), or I can input a new target from the keyboard. I then send this target to the ServoCat.
On the handpad variant, a goto, just gets the Nexus stored target and sends it to the ServoCat.

The serial link from Nexus to ServoCat remains. It provides the ServoCat with Az & Alt tracking rates.

[ServoCat setup.pdf](https://github.com/WimDeMeester/eFinder/files/12185594/ServoCat.setup.pdf)

**Reply by WimDeMeester** [reply]
Thanks, this makes sense!  So, I guess the only disadvantage of this is that if the eFinder crashes, I can no longer use the Nexus to directly drive the ServoCat, is that correct?
What I still don't understand is why the Nexus can only connect to the eFinder (in the normal case) and not to other applications, such as SkySafari using Wifi.  Isn't that exactly what we are doing at this moment?  I connect SkySafari to the Nexus Wifi and I also connect the eFinder to the wifi of the Nexus (well, now I'm using a external router, so I connect everything to the router). 
Can I have access to version 20 of the eFinder?  I can test this solution if you want to.

**Reply by AstroKeith** [reply]
Hmmm. If eFinder crashes, just do a reboot? It doesnt affect the Nexus DSC alignment, so you can get going again in a few seconds. If its a melt down, then replug the USB from Nexus to ServoCat (and change back the USB protocol from LX200 to ServoCat).

The problem with using the Nexus wifi is data request clashes. If SkySafari and eFinder are both connected via an external router, they are still both trying to access the nexus DSC on port 4060. Both are requesting RA & Dec data at quite a high frequency and it's not long before they clash.

I use the external wifi router to allow me to VNC into the eFinder.

The first user of ver20 wants to be involved in the code development (learning experience!). Currently ver20 works on my test rig but is a long way from being robust. He should be starting field testing when the longer nights return. It's also a big departure from your's and Mike build!

**Reply by WimDeMeester** [reply]
Indeed a reboot should work most of the time and replug the USB from the Nexus to servoCat is the option I missed...

**Comment by WimDeMeester** [comment]
Hi @AstroKeith,

I'm still a bit confused...  I try to explain my confusion 😛 

This is my configuration at this moment.  But if I understand you correctly, this will not work reliably.  It would work in one of these two cases:
- SkySafari is not used, or can be used, but not when it is connected to the telescope, so there will be no live updates of where the telescope is pointed.
- If version 20 of the eFinder is used, and the USB connection from the ServoCAT is connected directly to the eFinder, as well as the Nexus.  

Or will these clashes only occur when the VNC GUI is asking the Nexus constantly for the coordinates?
If the Nexus is connected to the eFinder via USB, the Nexus WiFi is only used for SkySafari, so I guess the tablet (and the Nexus) talk directly to each other, and don't need to be connected to the router.  Is this correct?  Does the eFinder need a Wifi connection at all in that case?  And is the travel router of any use?

![image](https://github.com/WimDeMeester/eFinder/assets/7261529/eaac0d05-2ac9-4dbf-8546-52ce12b977c4)

**Comment by WimDeMeester** [comment]
And will this configuration work with the Nexus DSC (not Pro)?

**Comment by AstroKeith** [comment]
Hi,

I think your explanation is correct… Answers in yellow highlight.


This is my configuration at this moment. But if I understand you correctly, this will not work reliably. It would work in one of these two cases:

SkySafari is not used, or can be used, but not when it is connected to the telescope, so there will be no live updates of where the telescope is pointed. Correct
If version 20 of the eFinder is used, and the USB connection from the ServoCAT is connected directly to the eFinder, as well as the Nexus. Correct, USB from eFinder to ServoCat and Serial from Nexus to ServoCAt
Or will these clashes only occur when the VNC GUI is asking the Nexus constantly for the coordinates? Clashes will eventually occur with both variants of eFinder.
If the Nexus is connected to the eFinder via USB, the Nexus WiFi is only used for SkySafari, so I guess the tablet (and the Nexus) talk directly to each other, and don't need to be connected to the router. Is this correct? Does the eFinder need a Wifi connection at all in that case? And is the travel router of any use? If you want to connect your tablet to the eFinder (for the GUI) then for the Nexus DSC Pro, you need a travel router. For the non Pro, the Nexus can act as a router.

Does this help?

Keith

**Comment by WimDeMeester** [comment]
Thanks, this does help.  So as far as I see, there are two possible configurations when using a Nexus DSC Pro: 
- If SkySafari is not used: 
![NexusDscPro](https://github.com/WimDeMeester/eFinder/assets/7261529/5cfb1202-fa65-46b4-805f-c0268d8c4a8c)
- If SkySafari is used:
![NexusDscProSkySafari](https://github.com/WimDeMeester/eFinder/assets/7261529/0b54a22d-bc3f-4afc-b73e-c2dc989ed069)

But in the last option, version 20 of the eFinder software is needed.  

I still have a few questions:
- If a Nexus is used (not the Nexus Pro), together with the ServoCat (and SkySafari), I guess also version 20 of the software is needed.  Is that correct?  The only difference is that the travel router can be removed from the images. 
- Was there ever a version 19 of the eFinder software?  I only have access up to version 18.
- Do you need an extra tester for version 20?  My secondary is away to the receive a new coating, but I guess I can test the eFinder, without being able to look through my telescope.

**Comment by AstroKeith** [comment]
Thats right.

I’ve copied ver20 to the google share.

Another option, would be like your first one, but with SkySafari connected to a port on the Pi. This would need a thread running to keep SkySafari happy. This is part solved by the ‘missing ver19’! Currently ver19 feeds RA & Dec to SkySafari, but doesnt take any inputs.

Keith

**Comment by AstroKeith** [comment]
Hi Wim,

I’ve uploaded ver 19 to the share, but restricted it to just you for now - although feel free to pass to Mike.
Ver19 is the ’no encoder’ push to version.

It is working fairly stably and so I am happy to share. It might be useful for you to see how I am serving the SkySafari interface. I tried to put this into a separate class but got hung up on threads not sharing data. So it currently using in the main eFinder code as lx200Socket(). SkySafari is very sensitive to timing and what it receives, so change at your peril!

A gps USB dongle is needed to replace the set up data received from the Nexus DSC. This requires gps3 and gpsd to be installed….

install gps3
	sudo -H pip install gps3
install gpsd
	sudo apt-get update
	sudo apt-get install gpsd gpsd-clients

I’ve also placed in the share usbhunt.py . This is my first attempt at scanning the Pi’s USB physical ports and finding what physical device is connected to which dev/ttyXXXX.
The Pi OS seems a bit arbitrary as to how it assigns these devices and I’m moving towards using this method for all my code variants. This is especially needed if someone plugs in a USB device after boot has finished. I’ll rewrite it as a general purpose class to be called at start up.

Cheers
Keith

**Comment by WimDeMeester** [comment]
Thanks, I will have a look at version 20 later (this week hopefully).  Is it already a bit stable?  What should still be done for version 20?

**Comment by AstroKeith** [comment]
Not having a ServoCat, I made an emulator. But I’m not able to verify my emulator represents a ServoCat well enough!

On my emulator ver20 works fine.

Forgot to mention, ver 20 includes a new feature. When a goto has been requested, it then watches for RA & Dec to stop moving, which signals the goto has finished. Then a new solve is done to show final position/accuracy. This is kind of built in to my ScopeDog systems, so needed a new approach for ServoCat as it doesnt give any indications it self. The new functions are gotoStopped() and getRadec()

Comments welcome!

Keith

**Reply by WimDeMeester** [reply]
Thanks!  I have some questions:
- I guess this version only works for the ServoCAT, not for other systems (like your ScopeDog).  Is this correct?
- Is this what is happening?
   - The telescope moves to a new location, after it receives the new coordinates from SkySafari or the Nexus.
   - When the telescope arrives at the new location, the eFinder takes an image and sends the offset to the ServoCAT.  Do I have to do this manually by clicking a button on the GUI or the handbox, or will this happen automatically?  I guess this can be done automatically, no?
   - A new solve is done automatically to show the final position / accuracy.

**Comment by AstroKeith** [comment]
Yes this is ServoCat only.

The basic functioning is……

The Serial Nexus to ServoCat cable tells ServoCat where it is, and what rate to track in Az & Alt. No Goto, moves or slews.

The USB socket on the ServoCat accepts the goto destination, and stop goto. Since ver7 of ServoCat it also accepts move (ie in response to arrow keys on SS or Nexus). As soon as the ServoCat sees a valid goto destination, it goes there.

Hence the eFinder now is in the USB path from the nexus to the ServoCat. eFinder can tell the ServoCat to do a goto (and stop). On the GUI the goto’s can be entered from the GUI, or downloaded from the Nexus (ie the goto destination that would have been sent if the eFinder wasnt in the loop). On the handpad the goto sent is the one held in the the Nexus.

On ver 20, the goto command from the eFinder, first does a local sync (to Nexus) then sends the destination coordinates (to ServoCat). Thus if the goto is a long distance, it might need to be repeated since the local sync is at the start of the slew not the end. The user it is aimed at wants to control the process and so it doesnt currently have the local sync at the end of the goto - but could be easily added as an option. (Its an option on ScopeDog)

So on reflection, I’m generating displays of ‘Goto++’ whereas it should be ‘Goto’ as the adjustment isnt automatic.

In all cases, goto++ docent send an offset to the drive to correct pointing. It does a local sync, and then repeats the goto.

Hope that’s clearer!

Keith

**Reply by WimDeMeester** [reply]
Thanks!  It might be clear tonight, so I might be able to test this.  I'll try to implement the local sync at the end of the goto as well, as that will fit my use case better.

**Comment by WimDeMeester** [comment]
Hi Keith,

I think there is still an error in the following image...  There should be a USB connection from the Nexus DSC Pro to the eFinder.  Is this correct? 

![NexusDscProSkySafari](https://github.com/WimDeMeester/eFinder/assets/7261529/56641790-519e-4240-84c7-48b8259a93a8)

Do I still need to configure something in the software (as the eFinder now has three different USB devices attached: the Nexus DSC Pro, the hand box and the ServoCAT)?

**Comment by AstroKeith** [comment]
Ooops, you’re right.

As the Nexus DSC USB is a host, you need a ttl to USB adapter cable. One like https://www.amazon.co.uk/gp/product/B08ZS4PHNL/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1

You dont need the positive wire connected, it gets its power from the USB plug end. Connect to the PI GPIO UART Tx Rx as described  on my website.

**Reply by WimDeMeester** [reply]
Are there specific settings needed in the Nexus DSC Pro for the USB protocol (as there are settings needed for the Wifi)?

**Comment by WimDeMeester** [comment]
Hi Keith,

Are you pointing me to this table? 
![image](https://github.com/WimDeMeester/eFinder/assets/7261529/b909a411-05d4-4e60-ac6c-d3f773b4207c)

I guess I will not be able to test everything tonight then, as I still need to order the cable...

**Comment by WimDeMeester** [comment]
So, this is the correct configuration then?
![NexusDscProSkySafari](https://github.com/WimDeMeester/eFinder/assets/7261529/720ed08a-6db0-4e8a-9c71-7d3e8cd34a66)

**Comment by AstroKeith** [comment]
Yes, that’s the table. Sorry, thought you had that installed!

It does seem that there are two Chinese factories producing these cables, and they reverse the white and green. Not always consistent either. It doesnt seem to harm anything by having them reversed but if you get no comms, then try th other way!

**Comment by AstroKeith** [comment]
You’ll just need to change the USB from ServoCat to LX200,9600,N,1

**Comment by WimDeMeester** [comment]
I tried to set up the ramdisk, but I get the following error message:
`mount: /var/tmp: wrong fs type, bad option, bad superblock on tmpfs, missing codepage or helper program, or other error.`

Any idea what I'm doing wrong?

**Comment by AstroKeith** [comment]
I may have been lax in my instructions - for clarity

Edit /fstab to add the line
Sudo nano /etc/fstab
	tmpfs /var/tmp tmpfs nodev,nosuid,size=10MB 0 0
	
Save & exit

Execute in termianl
sudo mount -a

Execute in terminal to check its done
df -h

**Reply by WimDeMeester** [reply]
That is exactly what I did, but the result is the error I mentioned before...

**Reply by WimDeMeester** [reply]
I found the solution!  The line should be 

tmpfs /var/tmp tmpfs nodev,nosuid,size=10M 0 0

So, for setting the size, we should use M, and not MB.

**Comment by AstroKeith** [comment]
Are you using the standard 32bit OS, or a 64bit?

**Comment by AstroKeith** [comment]
Well done!

**Comment by AstroKeith** [comment]
I’ve started ver 21 !!

This brings together (for me) all options and enhancements. So choice of drive (scope dog/servocat), image folder (sd card or ramdisk) etc will be determined by the efinder.config file.

I’ve tested quite a lot, so feel free to try it (instead of ver 20?) It’s in the share.

Also in that folder is usbhunt.py - I would be really interested if you could run that with a ServoCat connected by USB to the Pi. It should be a standard ‘USB printer cable’ ie USB A to USB B. This utility scans all the USB ports and reports back what is connected to what. I’d like to see what the ServoCat reports back! I can then finish the new usbAssign Class.

Notes:

ASICapture2.py is new. It takes the image folder destination as an argument.

Dont save index files to the same ramdisk folder as the image files. This upsets the plate solver. In fact there is no improvement in speed to have by moving index files to ram, as the OS does this anyway.

Look at efinder.config to see the new line entries.

Main-ef1_2.py is a new micro python code for the handset. It catches comms errors and stops the handset crashing (so far!). Once loaded, to stop the code (ie for re-loading etc) plug in the handset with the OK button pressed.

Cheers
Keith

**Reply by WimDeMeester** [reply]
Thanks! I will certainly test the new version and send the output of the usbhunt.py.  At the moment, I'm on holidays. When I return on August 28, I will start testing.

**Comment by AstroKeith** [comment]
Hi,

I’ve just completed a new version for someone on the Kansas Prairies!

No encoders, no display. Just solves and feeds SkySafari. His P is a hotspot so very easy to connect to.

Attached is the readme for more info. It’s in the share as ver 22. I had to change quite a lot to compensate for the lack of handpad and Nexus!

Keith

**Reply by WimDeMeester** [reply]
Hi Keith,
Is this working (solving) fast enough?  Is the setup using a motor driver, or is it just a push to?

**Comment by AstroKeith** [comment]
﻿﻿Yes. It’s a push too. The user (Duane in Kansas) wanted the 10 seconds cycle. Interestingly using the pi to generate ahotspot really seemed to clobber its responsiveness. Sent from my iPhoneOn 29 Aug 2023, at 10:08, Wim De Meester ***@***.***> wrote:﻿
Hi Keith,
Is this working (solving) fast enough?  Is the setup using a motor driver, or is it just a push to?

**Comment by AstroKeith** [comment]
Hi allAttached is the welcome message that Bentley Ousley has sent for the new eFinder discussion forum he created. A great idea. I think we’ve got about 50 users now worldwide. I’ll be posting some news when I get back from my short break. KeithSent from my iPhoneBegin forwarded message:From: Bentley Ousley ***@***.***>Date: 28 August 2023 at 17:29:48 BSTTo: ***@***.***: [eFinder] Welcome to the forumReply-To: ***@***.***﻿
I thought it might be nice to have a resource to discuss our usage experiences with Keith's eFinder system. Allow me to introduce myself and cover my use case and desires for using this system.
 
My name is Bentley Ousley (yes, just like my screen name). I've been involved in amateur astronomy for about 30 years. My main scope is a 20" f/3.3 New Moon Telescopes Dob with Nexus DSC/ServoCat installed. I began exploring plate solving as a way to correct the alignment and tracking issues I experienced with ServoCat. It seemed I was always correcting the position manually with ServoCat and longed for something better. I first tried using the Stellarmate system with an eye toward scripting the existing software system to meet my needs. I had some success with getting the system to adjust the scope, but it was obvious the changes required to make this work consistently would require a much deeper dive into the code.
 
I noticed Keith's work on a thread in the CloudyNights forum and thought, "gee, this guy is already most of the way there. "  Keith was kind enough to share his work with me and I've seen several iterations of the system. I'm currently making the changes needed to use V21 and am excited to see a method to eliminate the need to communicate between Nexus DSC and eFinder through WIFI. One of the issues that has kept me from using this system with any regularity has been the number of WIFI disconnection issues. I'm looking forward to making this system part of my regular observing set-up.
 
I look forward to hearing from other eFinder users and hope this forum will be helpful for sharing issues and solutions.
-Bentley Ousley
 

 


_._,_._,_



Groups.io Links:


  
    You receive all messages sent to this group.
  
  



View/Reply Online (#1) |


  Reply To Group
  
    | Reply To Sender
  

|

  Mute This Topic


| New Topic






Your Subscription |
Contact Group Owner |

Unsubscribe

 ***@***.***
_._,_._,_
